### ISSUE TITLE
glide 12.1 does not fetch git tag in cache

### ISSUE THREAD

**Issue by Cedric-Venet** [Masterminds/glide#604]
It seems like glide update the cache with only `git fetch` and not `git fetch --tags`. At least after pushing a new tagged version of a dependency, glide update did not see it. going in the cache, the new commit had been fetched but not its tag. after fetching the tags manually, glide update worked as expected.

**Comment by sdboyer** [comment]
interesting. yes, you're definitely right that it only runs `git fetch`, not `git fetch --tags`.  i suspect this hasn't been reported before because of the default `git fetch` mechanics:

> By default, any tag that points into the histories being fetched is also fetched; the effect is to fetch tags that point at branches that you are interested in. This default behavior can be changed by using the --tags or --no-tags options or by configuring remote.<name>.tagOpt. By using a refspec that fetches tags explicitly, you can fetch tags that do not point into branches you are interested in as well.

So, most repos probably tag on-branch, which means the tags have been picked up. Find a repo that tags off-branch, though, and this problem will arise.

**Comment by Cedric-Venet** [comment]
Oh yes sorry, looking at my library repo, it is indeed a strange state with a tag outside any branch. I have no idea how I was able to push a tag without pushing the branch with it... You can close the issue as its not something which should happen in practice.

**Comment by mattfarina** [comment]
I filed a bug on the VCS project so we can fix the issue there.

@sdboyer will fixing it on VCS cause that fix to roll into GPS properly?

**Comment by sdboyer** [comment]
@mattfarina yep yep - gps calls out to `Masterminds/vcs` for the `Update()` action, so it'll get the fix as soon as the dep itself is updated.
